lib/fs.js

```diff
@@ -1579,6 +1579,17 @@ function watch(filename, options, listener) {
   if (listener) {
     watcher.addListener('change', listener);
   }
+  if (options.signal) {
+    if (options.signal.aborted) {
+      process.nextTick(() => watcher.close());
+    } else {
+      const listener = () => watcher.close();
```

Comment:
nit: you could move the `listener` declaration above and pass it in `process.nextTick` to avoid code duplication.

Reply:
I actually find this way more readable though I don't have very strong feelings about it.